ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -1781,6 +1787,14 @@ private long computeProjectionSize(List<OrcProto.Type> fileTypes,
         " reader schema " + (readerSchema == null ? "NULL" : readerSchema.toString()) +
         " ACID scan property " + isAcidTableScan);
     }
+    boolean metricsEnabled = HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_SERVER2_METRICS_ENABLED);
+    long checkThresholdInSec = HiveConf.getTimeVar(conf,
+        HiveConf.ConfVars.HIVE_TXN_ACID_METRICS_DELTA_CHECK_THRESHOLD, TimeUnit.SECONDS);
+    float deltaPctThreshold = HiveConf.getFloatVar(conf, HiveConf.ConfVars.HIVE_TXN_ACID_METRICS_DELTA_PCT_THRESHOLD);
+
+    Map<String, Map<String, Integer>> deltas = new HashMap<>();
+    Stream.of(NUM_OBSOLETE_DELTAS, NUM_DELTAS, NUM_SMALL_DELTAS).forEach(
```

Comment:
Should this only be executed if metrics are enabled?

Reply:
fixed